### ISSUE TITLE
neovim can't find ruby host

### ISSUE THREAD

**Issue by shadow-absorber** [neovim/neovim#29553]
### Problem

neovim fails to find ruby host using checkhealth and ruby command

### Steps to reproduce

run `checkhealth ruby`
see this
```
provider.ruby: require("provider.ruby.health").check()

Ruby provider (optional) ~
- Ruby: ruby 3.0.6p216 (2023-03-30 revision 23a532679b) [x86_64-linux]
- WARNING `neovim-ruby-host` not found.
  - ADVICE:
    - Run `gem install neovim` to ensure the neovim RubyGem is installed.
    - Run `gem environment` to ensure the gem bin directory is in $PATH.
    - If you are using rvm/rbenv/chruby, try "rehashing".
    - See :help |g:ruby_host_prog| for non-standard gem installations.
    - You may disable this provider (and warning) by adding `let g:loaded_ruby_provider = 0` to your init.vim
    ```
  here is the output of `gem environment`
  
  ```
  RubyGems Environment:
  - RUBYGEMS VERSION: 3.3.25
  - RUBY VERSION: 3.0.6 (2023-03-30 patchlevel 216) [x86_64-linux]
  - INSTALLATION DIRECTORY: /usr/lib/ruby/gems/3.0.0
  - USER INSTALLATION DIRECTORY: /home/user/.local/share/gem/ruby/3.0.0
  - RUBY EXECUTABLE: /usr/bin/ruby
  - GIT EXECUTABLE: /usr/bin/git
  - EXECUTABLE DIRECTORY: /usr/bin
  - SPEC CACHE DIRECTORY: /home/user/.local/share/gem/specs
  - SYSTEM CONFIGURATION DIRECTORY: /etc
  - RUBYGEMS PLATFORMS:
     - ruby
     - x86_64-linux
  - GEM PATHS:
     - /usr/lib/ruby/gems/3.0.0
     - /home/user/.local/share/gem/ruby/3.0.0
  - GEM CONFIGURATION:
     - :update_sources => true
     - :verbose => true
     - :backtrace => true
     - :bulk_threshold => 1000
     - "gem" => "--user-install"
  - REMOTE SOURCES:
     - https://rubygems.org/
  - SHELL PATH:
     - /opt/devkitpro/tools/bin
     - /home/user/perl5/bin
     - /home/user/.local/share/luarocks/bin
     - /home/user/.config/lf
     - /home/user/.config/scripts
     - /home/user/.local/bin
     - /home/user/.local/share/pyenv/shims
     - /home/user/.npm/npm-global/bin
     - /usr/local/bin
     - /usr/bin
     - /bin
     - /usr/local/sbin
     - /opt/cuda/bin
     - /opt/cuda/nsight_compute
     - /opt/cuda/nsight_systems/bin
     - /var/lib/flatpak/exports/bin
     - /usr/lib/jvm/default/bin
     - /usr/bin/site_perl
     - /usr/bin/vendor_perl
     - /usr/bin/core_perl
     - /home/user/.local/share/gem/ruby/3.0.0/bin
  ```
  

### Expected behavior

for neovim to find the ruby host and be able to run ruby commands in command mode

### Neovim version (nvim -v)

NVIM v0.10.0

### Vim (not Nvim) behaves the same?

vim does not seem to have check health

### Operating system/version

endeavour os

### Terminal name/version

alacritty

### $TERM environment variable

tmux-256color

### Installation

pacman -S neovim

**Comment by Sebbito** [comment]
Same thing here.
I followed the advice from [this issue](https://github.com/neovim/neovim/issues/7730#issuecomment-352216451) and set up ruby provider in `init.lua` but even then it doesn't recognize it.

```lua
vim.g.ruby_host_prog = '/home/user/bin/neovim-ruby-host'
print (vim.g.ruby_host_prog) -- for debug
```

Opening neovim yields:

```
/home/user/bin/neovim-ruby-host
```

`echo g:ruby_host_prog` also yields the correct path but the checkhealth message does not change.

**Comment by shadow-absorber** [comment]
also saw that issue and tried the ruby provider command that was in there but that did also not work

**Comment by Gerg-L** [comment]
I can reproduce this issue with nixpkgs version of neovim

**Comment by Sebbito** [comment]
Spend some time bisecting this problem.
The bisect leads to b280d57.
Seems to have been broken since then.

**Comment by clason** [comment]
Can you please try the latest nightly? There was a fix to the ruby provider checkhealth on master (and backported to the 0.10 branch).

**Comment by Sebbito** [comment]
Theres still an error. A different one this time tho:

```
Ruby provider (optional)
- Ruby: ruby 3.3.2 (2024-05-30 revision e5a195edf6) [x86_64-linux]
- ERROR Failed to run healthcheck for "provider" plugin. Exception:
  Vim:E117: unknown function: provider#ruby#Detect
```

**Comment by clason** [comment]
Try with `nvim --clean`. How did you install the nightly?

**Comment by clason** [comment]
Yeah, that won't work. If runtime files are _removed_ in a commit, they are not automatically uninstalled. `rm -rf` the `/usr/local/share/nvim/runtime` directory and install again.

(Don't forget to do that when switching back to release, too!)

**Comment by Sebbito** [comment]
That's true! After manually rebuilding it now recognizes the provider!
Thanks a lot :)

**Comment by clason** [comment]
Great; the fix will be in the next release (0.10.1).
